src/scripts/picard/analysis/insertSizeHistogram.R

```diff
@@ -45,54 +44,53 @@ if (any(duplicated(headers))) {
       levels[length(levels)+1] <- headers[i]
     }
   }
+}
 
-  pdf(pdfFile)
-
-  for (i in 1:length(levels)) {
-    ## Reconstitutes the histogram column headers for this level
-    fr <- paste(levels[i], "fr_count", sep=".")
-    rf <- paste(levels[i], "rf_count", sep=".")
-    tandem <- paste(levels[i], "tandem_count", sep=".")
+pdf(pdfFile)
 
-    frrange = ifelse(fr %in% names(histogram), max(histogram[fr]), 0)
-    rfrange = ifelse(rf %in% names(histogram), max(histogram[rf]), 0)
-    tandemrange = ifelse(tandem %in% names(histogram), max(histogram[tandem]), 0)
+for (i in 1:length(levels)) {
+  ## Reconstitutes the histogram column headers for this level
+  fr <- paste(levels[i], "fr_count", sep=".")
+  rf <- paste(levels[i], "rf_count", sep=".")
+  tandem <- paste(levels[i], "tandem_count", sep=".")
 
-    yrange <- max(frrange, rfrange, tandemrange)
-    xrange <- ifelse(histoWidth > 0, histoWidth, max(histogram$insert_size))
+  frrange = ifelse(fr %in% names(histogram), max(histogram[fr]), 0)
+  rfrange = ifelse(rf %in% names(histogram), max(histogram[rf]), 0)
+  tandemrange = ifelse(tandem %in% names(histogram), max(histogram[tandem]), 0)
 
-    plot(x=NULL, y=NULL,
-         type="n",
-         main=paste("Insert Size Histogram for", levels[i], "\nin file", bamName),
-         xlab="Insert Size",
-         ylab="Count",
-         xlim=range(0, xrange),
-         ylim=range(0, yrange))
+  yrange <- max(frrange, rfrange, tandemrange)
+  xrange <- ifelse(histoWidth > 0, histoWidth, max(histogram$insert_size))
 
-    colors <- c()
-    labels <- c()
+  plot(x=NULL, y=NULL,
+       type="n",
+       main=paste("Insert Size Histogram for", levels[i], "\nin file", bamName),
+       xlab="Insert Size",
+       ylab="Count",
+       xlim=range(0, xrange),
+       ylim=range(0, yrange))
 
-    if (fr %in% names(histogram) ) {
-      lines(histogram$insert_size, as.matrix(histogram[fr]),  type="h", col="red")
-      colors <- c(colors, "red")
-      labels <- c(labels, "FR")
-    }
-    if (rf %in% names(histogram)) {
-      lines(histogram$insert_size, as.matrix(histogram[rf]),  type="h", col="blue")
-      colors <- c(colors, "blue")
-      labels <- c(labels, "RF")
-    }
+  colors <- c()
+  labels <- c()
 
-    if (tandem %in% names(histogram)) {
-      lines(histogram$insert_size, as.matrix(histogram[tandem]),  type="h", col="orange")
-      colors <- c(colors, "orange")
-      labels <- c(labels, "TANDEM")
-    }
+  if (fr %in% names(histogram) ) {
```

Comment:
if I were you, I'd complain about the inconsistent spaces in this if statement and the following ones...:-)